students/briggsm/session04/mb_trigram.py

```diff
@@ -1,54 +1,81 @@
+''''
+Matt Briggs Trigram Exercise
+Version 2.0 2017.11.19
+'''
+
 import random as rand
 
+
 def create_corpus(infile):
-    '''Input: a filename
-       Output: a list of words.'''
+    ''' Input: a filename of a text file
+        Output: a list of words from the text file.'''
     corpus = []
     with open(infile) as lines:
         lines.readline()
         for line in lines:
             line = line.strip()
             words = line.split(" ")
-            for i in range(len(words)):
-                corpus.append(words[i])
+            for word in enumerate(words):
+                corpus.append(word[1])
     return corpus
 
+
 def create_trigram(textlist):
-    '''Input: a list of words from a text body.
-       Output: a dictionary of two words in sequence and the word after.'''
+    ''' Input: a list of words from a text body.
+        Output: a dictionary of two words in sequence and the word after.'''
     trigram = {}
-    for w in range(len(textlist[:-2])):
-        firstword = textlist[w]
-        secondword = textlist[w+1]
-        thirdword = textlist[w+2]
-        trigram[thirdword] = (firstword, secondword)
+    for x in range(len(textlist[:-2])):
+        firstword = textlist[x]
+        secondword = textlist[x+1]
+        thirdword = textlist[x+2]
+        #print("[({},{}] = {}".format(firstword, secondword, thirdword))
+        if (firstword, secondword) in trigram:
+            trigram[(firstword, secondword)].append(thirdword)
+        else:
+            trigram[(firstword, secondword)] = [thirdword]
     return trigram
 
+
 def get_values(trigram):
     '''Input: a dictionary.
-       Output: a list of keys.'''
+       Output: a list of word pair keys.'''
     words = []
     for w in trigram.keys():
```

Comment:
this is just:

`list(trigram)`

:-)

But think about if you need to create a separate list of what's already in the dict. You need it for making a random choice from it, but I think that's it, so maybe just create the listwhen you need it:

`seed = random.choice(list(trigrams))`

(untested -- something like that)

It's generally a bid idea to have a copy of the same info in two places...